### PR TITLE
Added tests to take into account the _disablenotif option

### LINKED ISSUE #1229: '_no_notif' has disapeared from all source code
Hello,

In 0.83.8, I used this flag to prevent sending notifications in some particular cases.

Ex: a requester is creating a ticket, and a plugin is creating a task (assigned to the requester) to keep track of some information during creation, and as this task is done, the requester will receive three notifications (one for ticket creation, one for task assignment, and one for task update).

This option was in the code in the following files:
- commondbchild.class.php
- commondbrelation.class.php
- commonitilactor.class.php
- commonitiltask.class.php
- problem.class.php
- problem_ticket.class.php
- ticket.class.php
- ticketfollowup.class.php

and in some plugins (not exhaustive list):
- behaviors\inc\ticket_user.class.php

Where is it gone, and why?
Is there any possibilities to call `CommonITILTask::add()` or `CommonITILTask::update()` and to pass a parameter to prevent notifications?

Or should we use a `$CFG_GLPI["use_mailing"] = false;` before add or update, and after them a `$CFG_GLPI["use_mailing"] = $oldvalue;` to restore former value?

Thank you,
Regards,
Tomolimo

### PR DESCRIPTION
**Pull request by tomolimo**
which should prevent notifications when present in the $this->input of the current object
Fixes #1229

### CODE REVIEW

File: inc/problem.class.php [hunk 1]
```diff
@@ -164,8 +164,15 @@ function canCreateItem() {
...
    function pre_deleteItem() {
+      global $CFG_GLPI;
 
-      NotificationEvent::raiseEvent('delete', $this);
+      $donotif = $CFG_GLPI["use_mailing"];
+      if (isset($this->input['_disablenotif'])) {
+         $donotif = false;
+      }
+      if( $donotif ) {
+         NotificationEvent::raiseEvent('delete', $this);
```
Comment (orthagh) [inline inc/problem.class.php]: Why not using the same syntax as before, like:
```php
if (!isset($this->input['_disablenotif']) && $CFG_GLPI["use_mailing"]) {
```

File: inc/project.class.php [hunk 2]
```diff
@@ -280,8 +280,15 @@ function post_getFromDB() {
 
 
    function pre_deleteItem() {
+      global $CFG_GLPI;
 
-      NotificationEvent::raiseEvent('delete',$this);
+      $donotif = $CFG_GLPI["use_mailing"];
+      if (isset($this->input['_disablenotif'])) {
```
Comment (orthagh) [inline inc/project.class.php]: same for here

File: inc/projecttask.class.php [hunk 3]
```diff
@@ -257,8 +257,15 @@ function getTeamCount() {
 
 
    function pre_deleteItem() {
+      global $CFG_GLPI;
 
-      NotificationEvent::raiseEvent('delete',$this);
+      $donotif = $CFG_GLPI["use_mailing"];
+      if (isset($this->input['_disablenotif'])) {
+         $donotif = false;
```
Comment (orthagh) [inline inc/projecttask.class.php]: same

File: inc/ticket.class.php [hunk 4]
```diff
@@ -519,8 +519,15 @@ function getDefaultActorRightSearch($type) {
 
 
    function pre_deleteItem() {
+      global $CFG_GLPI;
 
-      NotificationEvent::raiseEvent('delete',$this);
+      $donotif = $CFG_GLPI["use_mailing"];
+      if (isset($this->input['_disablenotif'])) {
+         $donotif = false;
```
Comment (orthagh) [inline inc/ticket.class.php]: same

**Comment by tomolimo** [comment]
Thank you :smiley:

**Pull request merged**